test/front_desk_test.rb

```diff
@@ -0,0 +1,304 @@
+require_relative 'test_helper'
+require 'date'
+
+describe "FrontDesk" do
+  describe "#initialize" do
+    it "Creates an instance of FrontDesk" do
+      front_desk = Hotel::FrontDesk.new
+      expect(front_desk).must_be_kind_of Hotel::FrontDesk
+    end
+
+    it "Keeps track of rooms" do
+      front_desk = Hotel::FrontDesk.new
+      expect(front_desk).must_respond_to :rooms
+    end
+
+    it "Keeps track of reservations" do
+      front_desk = Hotel::FrontDesk.new
+      expect(front_desk).must_respond_to :reservations
+    end
+
+    it "Creates an array of rooms 1 through 20" do
+      front_desk = Hotel::FrontDesk.new
+      expect(front_desk.rooms).must_be_kind_of Array
+      expect(front_desk.rooms.length).must_equal 20
+
+      front_desk.rooms.each do |room|
+        expect(room).must_be_kind_of Integer
+      end
+
+      expect(front_desk.rooms.first).must_equal 1
+      expect(front_desk.rooms.last).must_equal 20
+    end
+
+    it "Creates an empty array of reservations" do
+      front_desk = Hotel::FrontDesk.new
+      expect(front_desk.reservations).must_be_kind_of Array
+      expect(front_desk.reservations.length).must_equal 0
+    end
+  end
+
+  describe "#make_reservation" do
+    it "raises ArgumentError for passing invalid arguments" do
+      front_desk = Hotel::FrontDesk.new
+      expect{
+        front_desk.make_reservation("2020, 2, 12", Date.new(2020, 5, 1))
+      }.must_raise ArgumentError
+      
+    end
+
+    it "makes reservation" do 
+      front_desk = Hotel::FrontDesk.new
+      made_reservation = front_desk.make_reservation(Date.new(2020, 5, 1), Date.new(2020, 5, 4))
+      expect(made_reservation.last).must_be_kind_of Hotel::Reservation
+      expect(made_reservation.last.date_range.start_date).must_equal Date.new(2020, 5, 1)
+      expect(made_reservation.last.date_range.end_date).must_equal Date.new(2020, 5, 4)
+
+    end
+
+    it "adds a newly created reservation to the reservations list" do
+      front_desk = Hotel::FrontDesk.new
+      reservations_before = front_desk.reservations.length
+      puts "before length = #{reservations_before}"
```

Comment:
Please delete your `puts` statements before submitting your project :) It helps other people when they run your tests and code!